From d2b9c9f80c02d91392d3e756a8bab4c75a0fd668 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:24 -0400
Subject: [tools] perf/scripting/python: Shorten function signatures

Message-id: <1408953210-26343-197-git-send-email-jolsa@redhat.com>
Patchwork-id: 89035
O-Subject: [PATCH RHEL7.1 BZ1133083 196/322] perf scripting python: Shorten function signatures
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit b7fff6b5f977115be1757f18b1aca928803b1e17
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Thu Dec 19 16:34:52 2013 -0300

description
===========
Removing unused parameters.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/scripting-engines/trace-event-python.c b/tools/perf/util/scripting-engines/trace-event-python.c
index 53c20e7..fc00792 100644
--- a/tools/perf/util/scripting-engines/trace-event-python.c
+++ b/tools/perf/util/scripting-engines/trace-event-python.c
@@ -231,13 +231,10 @@ static inline struct event_format *find_cache_event(struct perf_evsel *evsel)
  return event;
 }
 
-static void python_process_tracepoint(union perf_event *perf_event
-          __maybe_unused,
-     struct perf_sample *sample,
-     struct perf_evsel *evsel,
-     struct machine *machine __maybe_unused,
-     struct thread *thread,
-     struct addr_location *al)
+static void python_process_tracepoint(struct perf_sample *sample,
+          struct perf_evsel *evsel,
+          struct thread *thread,
+          struct addr_location *al)
 {
  PyObject *handler, *retval, *context, *t, *obj, *dict = NULL;
  static char handler_name[256];
@@ -351,11 +348,8 @@ static void python_process_tracepoint(union perf_event *perf_event
  Py_DECREF(t);
 }
 
-static void python_process_general_event(union perf_event *perf_event
-      __maybe_unused,
-      struct perf_sample *sample,
+static void python_process_general_event(struct perf_sample *sample,
       struct perf_evsel *evsel,
-      struct machine *machine __maybe_unused,
       struct thread *thread,
       struct addr_location *al)
 {
@@ -411,22 +405,20 @@ exit:
  Py_DECREF(t);
 }
 
-static void python_process_event(union perf_event *perf_event,
+static void python_process_event(union perf_event *event __maybe_unused,
      struct perf_sample *sample,
      struct perf_evsel *evsel,
-     struct machine *machine,
+     struct machine *machine __maybe_unused,
      struct thread *thread,
      struct addr_location *al)
 {
  switch (evsel->attr.type) {
  case PERF_TYPE_TRACEPOINT:
-  python_process_tracepoint(perf_event, sample, evsel,
-       machine, thread, al);
+  python_process_tracepoint(sample, evsel, thread, al);
   break;
  /* Reserve for future process_hw/sw/raw APIs */
  default:
-  python_process_general_event(perf_event, sample, evsel,
-          machine, thread, al);
+  python_process_general_event(sample, evsel, thread, al);
  }
 }
 
-- 
1.7.1